.github/workflows/ci.yml

```
@@ -10,54 +10,32 @@ permissions:
   actions: read
   contents: read
 
-env:
-  POETRY_VERSION: "1.6.1"
-  PYTHON_VERSION: "3.11"
-
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
-        id: setup-python
-        with:
-          python-version: ${{ env.PYTHON_VERSION }}
+      - name: 🛎️ Checkout
+        uses: actions/checkout@v4
 
-      - name: Load cached Poetry installation
-        id: cached-poetry
-        uses: actions/cache@v4
-        with:
-          path: ~/.local
-          key: poetry-${{ runner.os }}-${{ env.POETRY_VERSION  }}-${{ steps.setup-python.outputs.python-version }}
-      - name: Install Poetry
-        id: install-poetry
-        if: steps.cached-poetry.outputs.cache-hit != 'true'
-        uses: snok/install-poetry@v1
-        with:
-          version: ${{ env.POETRY_VERSION }}
-          virtualenvs-create: true
-          virtualenvs-in-project: true
-          installer-parallel: true
-      - name: Poetry git plugin
-        run: poetry self add poetry-git-version-plugin
-      - name: Load cached venv
-        id: cached-poetry-dependencies
-        uses: actions/cache@v4
+      - name: 📦 Install uv
+        uses: astral-sh/setup-uv@v4
```

In the meantime this `setup-uv` action has gotten a `v5` release. 

Also, by explicitly setting a `python-version`, you avoid having to use a `.python-version` file

```suggestion
        uses: astral-sh/setup-uv@v5
        with:
          python-version: "3.12"
```

---

I think using a `.python-version` uses a standard way is a better way to declare compatible python version which I can refer to anywhere in the code, if I use it explicitly, I would have edit all the places where I referred to the incompatibility whenever I had to increment the python version.

---

I think using a file makes people to recognize the compatible version for a particular project